pac4j-saml-db/src/main/java/org/pac4j/saml/dbclient/dao/api/SamlClientDao.java

```diff
@@ -0,0 +1,22 @@
+package org.pac4j.saml.dbclient.dao.api;
+
+import java.util.List;
+
+import org.pac4j.saml.dbclient.DbLoadedSamlClientConfiguration;
+
+
+/**
+ * DAO to manipulate SAML Client Configurations stored in a data source.
+ * 
+ * @author jkacer
+ */
+public interface SamlClientDao {
+
+	/**
+	 * Loads all existing SAML Client Configurations.
+	 * 
+	 * @return A list of configurations.
+	 */
+	public List<DbLoadedSamlClientConfiguration> loadAllClients();

```

Comment:
Done. I have added a method to load a single client's configuration.

As for caching, I prefer not to have it in the base DAO implementation, so I created a very simple caching implementation of the DAO, which uses an underlying "real" DAO and only adds some trivial caching. Please see class CachingSamlClientDaoImpl.